cms/djangoapps/contentstore/utils.py

```diff
@@ -1523,6 +1523,46 @@ def format_in_process_course_view(uca):
     return active_courses, archived_courses, in_process_course_actions
 
 
+def get_course_context_v2(request):
+    """Get context of the homepage course tab from the Studio Home."""
+
+    from cms.djangoapps.contentstore.views.course import (
+        get_courses_accessible_to_user_v2,
+        ENABLE_GLOBAL_STAFF_OPTIMIZATION,
+    )
+
+    def format_in_process_course_view(uca):
+        """
+        Return a dict of the data which the view requires for each unsucceeded course.
+
+        Args:
+            uca: CourseRerunUIStateManager object.
+        """
+        return {
+            'display_name': uca.display_name,
+            'course_key': str(uca.course_key),
+            'org': uca.course_key.org,
+            'number': uca.course_key.course,
+            'run': uca.course_key.run,
+            'is_failed': uca.state == CourseRerunUIStateManager.State.FAILED,
+            'is_in_progress': uca.state == CourseRerunUIStateManager.State.IN_PROGRESS,
+            'dismiss_link': reverse_course_url(
+                'course_notifications_handler',
+                uca.course_key,
+                kwargs={
+                    'action_state_id': uca.id,
+                },
+            ) if uca.state == CourseRerunUIStateManager.State.FAILED else ''
+        }
+
+    optimization_enabled = GlobalStaff().has_user(request.user) and ENABLE_GLOBAL_STAFF_OPTIMIZATION.is_enabled()
+
+    org = request.GET.get('org', '') if optimization_enabled else None
```

Comment:
It seems like blank vs empty string will result in different behavior here? Can you add a comment explaining what the expected difference is between those two?  Or maybe that explaination needs to go into the docstring of `get_course_accessible_to_user_v2`?

Reply:
This is precisely how V1 behaves. So, I'll have to test out the behavior with those specifications. I'll make sure to answer once I know more.

Reply:
So I researched the behavior: `org=` and `org=''`, so for empty strings `"''"` this line would be executed: https://github.com/openedx/edx-platform/pull/34173/files#diff-937adee5129f53ead6b3c7c988d70db9e9e95862f17a6f85a1eea2388d12b461R443, but for blank `''` it will not. Unless there are organizations named `''` -from what I see, you can actually insert rows into the course overview table with the org `''` [since there are no restrictions from the course table](https://github.com/openedx/edx-platform/blob/43747ee8f08046a8892b5ad6cc9534fc0c2a896d/openedx/core/djangoapps/content/course_overviews/models.py#L75-L76)- then the result would be empty for the `"''"` case. 
I'm not sure why this is a case considered, so I can't explain it. I'll ask around though. I'll post here what I find.

Reply:
By the way, these helpers (get_course_context_v2, etc) are based on the ones used by the v1 API with some modifications:
get_course_context_v2 -> [based on get_course_context](https://github.com/openedx/edx-platform/blob/43747ee8f08046a8892b5ad6cc9534fc0c2a896d/cms/djangoapps/contentstore/utils.py#L1483-L1524)
get_courses_accessible_to_user_v2 -> [based on get_courses_accessible_to_user](https://github.com/openedx/edx-platform/blob/43747ee8f08046a8892b5ad6cc9534fc0c2a896d/cms/djangoapps/contentstore/views/course.py#L635-L657)
_accessible_courses_summary_iter_v2 -> [based on _accessible_courses_summary_iter](https://github.com/openedx/edx-platform/blob/43747ee8f08046a8892b5ad6cc9534fc0c2a896d/cms/djangoapps/contentstore/views/course.py#L394-L419)
I thought the transition would be easier this way instead of modifying the behavior of the existing ones in case other structures depended on them. Now I think it's even more confusing having it this way, duplicating code in a V2 version helper.